pkg/runtime/ecosystem/javascript.go

```diff
@@ -0,0 +1,66 @@
+package ecosystem
+
+import (
+	"path/filepath"
+
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/osutils"
+
+	"github.com/ActiveState/cli/pkg/buildplan"
+)
+
+const nodeModulesDir = "usr/lib/node_modules"
+
+type JavaScript struct {
+	runtimePath string
+	packages    []string
+}
+
+func (e *JavaScript) Init(runtimePath string, buildplan *buildplan.BuildPlan) error {
+	e.runtimePath = runtimePath
+	e.packages = []string{}
+	return nil
+}
+
+func (e *JavaScript) Namespaces() []string {
+	return []string{"language/javascript"}
+}
+
+func (e *JavaScript) Add(artifact *buildplan.Artifact, artifactSrcPath string) ([]string, error) {
+	files, err := fileutils.ListDir(artifactSrcPath, false)
+	if err != nil {
+		return nil, errs.Wrap(err, "Unable to read artifact source directory")
+	}
+	var packageName string
+	for _, file := range files {
+		ext := filepath.Ext(file.Name())
+		if ext != ".tar.gz" && ext != ".tgz" {
+			continue
+		}
+		packageName = file.Name()[:len(file.Name())-len(ext)]
+		e.packages = append(e.packages, file.AbsolutePath())
+	}
+	installedDir := filepath.Join(nodeModulesDir, packageName) // Apply() will install here
+	return []string{installedDir}, nil
+}
+
+func (e *JavaScript) Remove(artifact *buildplan.Artifact) error {
+	return nil // TODO: CP-956
+}
+
+func (e *JavaScript) Apply() error {
+	if len(e.packages) == 0 {
+		return nil // nothing to do
+	}
+
+	args := []string{"install", "-g", "--offline"}
```

Comment:
```suggestion
	args := []string{"install", "--offline"}
```

This installs it into the global package directory, which the local project will make use of. I imagine you should be able to just drop this.

Reply:
Okay, I've experimented with the `NPM_CONFIG_PREFIX` and it functions like Java's `CLASSPATH` such that we can have runtime-specific npm operations.

We still need `-g` because it will install to the runtime's _node_modules/_ directory instead of making a node_modules in the current working directory and installing to that.

Reply:
I'm not grokking from that why the `-g` flag is necessary? To me dropping `-g` would address that ticket?

For docker there are 100% going to be manual stitching steps for now. We shouldn't be forcing things in the State Tool that only address that use-case.

Reply:
If I drop `-g`, `state install` will create a _node_modules/_ folder in the current directory that I ran `state install` in. Is that desirable? I don't think so. I think we want the packages to be installed in the runtime cache's _node_modules/_ folder. To make that happen, we need the `-g`.

To put it another way, when you run `state install` from your project directory, do you want a node_modules/ to be there in that directory you've checked out your Platform project in? That's not how it is for our other runtimes. All of their dependencies are located in the runtime cache, which is what `-g` is doing.

Reply:
I see. Isn't that a matter of running it from a different working directory then? I mean the npm execution we do, not the `state install command` itself.

It might also necessitate we create a bogus `package.json`.

Either way I'd consider this in scope for this PR. ie. it's a requirement for this PR to install inside our virtual environment.